src/lifecycle.js

```diff
@@ -51,90 +52,141 @@ function parseEvent (e) {
-  } else if (typeof component.attributes === 'function') {
-    callback = component.attributes;
+/**
+* Sets the defined attributes to their default values, if specified.
+*
+* @param {Element} target The web component element.
+* @param {Object} component The web component definition.
+*
+* @returns {undefined}
+*/
+function initAttributes (target, component) {
+  var componentAttributes = component.attributes;
+
+  if (typeof componentAttributes !== 'object') {
+    return;
   }
 
-  // There may still not be a callback.
-  if (callback) {
-    callback(target, {
-      type: type,
-      name: name,
-      newValue: newValue,
-      oldValue: oldValue
-    });
+  for (var attribute in componentAttributes) {
+    if (hasOwn(componentAttributes, attribute) && hasOwn(componentAttributes[attribute], 'default') && !target.hasAttribute(attribute)) {
+      var value = componentAttributes[attribute].default;
+      value = typeof value === 'function' ? value(target) : value;
+      target.setAttribute(attribute, value);
+    }
   }
 }
 
-function triggerAttributesCreated (target, component) {
-  var a;
-  var attrs = target.attributes;
-  var attrsCopy = [];
-  var attrsLen = attrs.length;
+/**
+* Defines a property that proxies the specified attribute.
+*
+* @param {Element} target The web component element.
+* @param {String} attribute The attribute name to proxy.
+*
+* @returns {undefined}
+*/
+function defineAttributeProperty (target, attribute) {
+  Object.defineProperty(target, camelCase(attribute), {
+    get: function () {
+      return this.getAttribute(attribute);
+    },
+    set: function (value) {
+      if (value === undefined) {
+        this.removeAttribute(attribute);
+      } else {
+        this.setAttribute(attribute, value);
+      }
+    }
+  });
+}
 
-  for (a = 0; a < attrsLen; a++) {
-    attrsCopy.push(attrs[a]);
+/**
+* Adds links from attributes to properties.
+*
+* @param {Element} target The web component element.
+* @param {Object} component The web component definition.
+*
+* @returns {undefined}
+*/
+function addAttributeToPropertyLinks (target, component) {
+  var componentAttributes = component.attributes;
+
+  if (typeof componentAttributes !== 'object') {
+    return;
   }
 
-  // In default web components, attribute changes aren't triggered for
-  // attributes that already exist on an element when it is bound. This sucks
-  // when you want to reuse and separate code for attributes away from your
-  // lifecycle callbacks. Skate will initialise each attribute by calling the
-  // created callback for the attributes that already exist on the element.
-  for (a = 0; a < attrsLen; a++) {
-    var attr = attrsCopy[a];
-    triggerAttributeChanged(target, component, {
-      name: attr.nodeName,
-      newValue: attr.value || attr.nodeValue
-    });
+  for (var attribute in componentAttributes) {
+    if (hasOwn(componentAttributes, attribute) && !hasOwn(target, attribute)) {
```

Comment:
It just means that it doesn't need to set up the link for attributes that already have it such as `id`. If we didn't do this check, that would unnecessarily be overridden. If we threw an error, then that would mean they couldn't define any lifecycle behaviour for any existing attribute that has a corresponding property.

The original intent of this feature is to make attributes that don't have a corresponding property behave like those that do without the user needing to have any knowledge about which one's do and don't.